components/feature/session/src/main/java/mozilla/components/feature/session/SessionFeature.kt

```diff
@@ -51,6 +51,7 @@ class SessionFeature(
             engineView.clearSelection()
             return true
         } else if (session?.canGoBack == true) {
+            session.searchTerms = ""
```

Comment:
It doesn't make sense to me that this lives in `feature-session` since we would want that property to have the right state without the feature component in use.

Is that maybe something that should happen in `EngineObserver`?

Reply:
Okay, I'll try to fix it there, thanks.

Reply:
Fixed in [a new PR](https://github.com/mozilla-mobile/android-components/pull/6167) & new branch.